### PR TITLE
Add documentation on the design decisions

### PR DESCRIPTION
**Pull request by rowanmanning**
@JSRedondo suggested this based on the PR comments in #1. I decided to add as a separate document.

### CODE REVIEW

**Review by JSRedondo — APPROVED** [review]
Great!

File: docs/design.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  1. Have tooling installed in the top level package and run these tools against the whole repo
+
+  2. Have tooling installed in each package and have commands in the top level package to run these
+
+We have opted to have tooling installed at the top level. This speeds up build times, because Jest and ESLint have non-trivial start up times and multiplying this by the number of packages can lead to a lot of overhead. This also means that the tooling is consistent between all of the different packages and upgrading the tooling happens in one place. There is more information on [how to run tests](./contributing.md#testing) in the contributing guide.
+
+### Monorepo management
+
+We decided that [npm workspaces](https://docs.npmjs.com/cli/v8/using-npm/workspaces) and minimal tooling would be the best supported route forward. As we're opting for [testing at the top level](#testing), we have minimal need for running the same scripts across all packages. If we do want to then it's possible by adding the `--workspaces` flag to any npm command.
```
Comment (kavanagh) [inline docs/design.md:42]: I think future generation might be curious about the decision to have a monorepo.

File: docs/design.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+We made two language decisions when starting this project:
+
+  1. We decided to use JavaScript rather than TypeScript
+
+  2. We decided to use CommonJS modules rather than ES Modules
+
+The combination of the above allows us to work on and publish the packages in this monorepo without needing a build step. This means that the code we write is exactly the same as the code run inside our applications.
+
+In terms of TypeScript, the key benefits are having type safety and type hinting in your editor. We can achieve both of these without writing TypeScript. Using [JavaScript with JSDoc comments to document types](https://www.typescriptlang.org/docs/handbook/type-checking-javascript-files.html), VSCode (used by the majority of our engineers) offers the same level of type hinting as it does with TypeScript. It's also possible to run the TypeScript type checker against JavaScript code to verify that everything is type safe (e.g. using `tsc --checkJS`).
```
[the remark this answers is not in the document]
Comment (rowanmanning) [reply docs/design.md:23]: JSDoc comments work in dependents of the library too. Support for JSDoc is first-class in TypeScript and so to a consuming app they will get exactly the same experience as if we were writing TypeScript 👍

[the remark this answers is not in the document]
Comment (rowanmanning) [reply docs/design.md:23]: That includes types which are defined manually with JSDoc's `@typedef`, e.g:

```js
/**
 * @typedef {Object} ExampleConfig
 * @property {Array<String>} [exampleProp = []]
 * @property {Number} [anotherExample]
 */
```

FWIW I agree the experience for someone _using_ this library is more important than our experience building it, so making sure we properly define types etc is a definite priority 🙂

File: docs/design.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
+
+# Design Guide
+
+This document outlines the design decisions made in this project.
```
[the remark this answers is not in the document]
Comment (rowanmanning) [reply docs/design.md:4]: Yep definitely, I'll pop another task on the ticket to write it up

**Pull request merged**